### PR TITLE
Add Block str support for python templated strings

### PR DESCRIPTION
**Pull request by dvargasfuertes**
@coderabbitai summary

### CODE REVIEW

File: generators/python-v2/ast/src/TypeInstantiation.ts [hunk 1]
```diff
@@ -86,6 +91,10 @@ export class TypeInstantiation extends AstNode {
         return new this({ type: "str", value });
     }
 
+    public static blockStr(value: string): TypeInstantiation {
+        return new this({ type: "block_str", value });
+    }
+
```
Comment (amckinney) [inline generators/python-v2/ast/src/TypeInstantiation.ts:97]: Let's consolidate this functionality in the `str` helper above. Check out PHP's `multiline` option in [this method](https://github.com/fern-api/fern/blob/1908ee6716e2779e1af4a71865673397fb26f55b/generators/php/codegen/src/ast/Type.ts#L305), which prevents the caller from specifying `multiline` unless they actually care about it. That way there's a single `str` AST node, and it knows how to render itself based on the options provided.

I imagine we'd have something like:

```typescript
public static str(value: string, { block }: { block?: boolean } = {}): Type {
```

File: generators/python-v2/ast/src/TypeInstantiation.ts [hunk 2]
```diff
@@ -145,6 +154,11 @@ export class TypeInstantiation extends AstNode {
             case "str":
                 writer.write(`"${this.internalType.value}"`);
                 break;
+            case "block_str": {
+                const suffix = this.internalType.value.endsWith("\n") ? "\\" : "\n";
+                writer.write(`"""\\\n${this.internalType.value}${suffix}"""`);
```
Comment (amckinney) [inline generators/python-v2/ast/src/TypeInstantiation.ts:159]: This behavior would be handled in the `str` case in `if (this.internalType.block) { ... }`.

**Review by amckinney — APPROVED** [review]
Stamping to unblock, assuming that we move `block_str` into `str` as an option.

**Comment by dvargas92495** [comment]
Closing in favor of another previously merged pr by our team

**Pull request closed without merging**